includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -207,22 +207,6 @@ public function create_and_confirm_intention(
 		$request['description']    = $this->get_intent_description( $metadata['order_id'] ?? 0 );
 
 		$available_payment_methods = WC_Payments::get_gateway()->get_upe_available_payment_methods();
```

Comment:
```suggestion
		$available_payment_methods = WC_Payments::get_gateway()->get_upe_available_payment_methods();
		$request['payment_method_types'] = $available_payment_methods;
```
It looks like `$available_payment_methods` is unused in this function, however, I think it is still required.

I successfully completed a checkout with SEPA as my payment method and opted to save the payment method. When I tried to complete a new checkout using this saved payment method, the checkout threw an error. Checking the logs I discovered the following message.

> ERROR The PaymentMethod provided (sepa_debit) is not allowed for this PaymentIntent. Please attach a PaymentMethod of one of the following types: card. Alternatively, update the allowed payment_method_types for this PaymentIntent to include "sepa_debit". (invalid_request_error)

I think this is because `UPE_Payment_Gateway::process_payment` reroutes `process_payment` to the parent when using a saved payment method, which will ultimately get sent to this `create_and_confirm_intention` function. By default the array for `payment_method_types` only contains `card`; if we are using another payment method type, it needs to be added here. I think SEPA is the only other payment method that is reusable (i.e. that supports recurring payments and is saveable), so I think this is the only case that would cause this section to error out.

At a minimum, we need to add `sepa_debit` to the array of `payment_method_types` at this point. Off the top of my head, adding additional payment methods will not cause any problems, though, having said that I know that when checking out regularly through the UPE, just prior to confirming the intent, [we do update the payment intent to only include the selected payment method in the array of `payment_method_types`](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/wc-payment-api/class-wc-payments-api-client.php#L308-L311), so perhaps `payment_method_types` at this point should only contain the payment method type used (in this case, being `[ 'sepa_debit' ]`) instead of using all of the enabled payment method types inside `$available_payment_methods`.